### PR TITLE
Rename gnupg_key_id parameter to signing_keys

### LINKED ISSUE #139: key has changed since the start of the year
Note to future me, the key has changed... 

when you run this as root manually on any machine
```bash
curl -sSL https://rvm.io/mpapis.asc | gpg --import -
>>gpg: key 3804BB82D39DC0E3: public key "Michal Papis (RVM signing) <mpapis@gmail.com>" imported
>> gpg: Total number processed: 1
>> gpg:               imported: 1
>> gpg: no ultimately trusted keys found
```
looking at the source it has
```puppet
$gnupg_key_id = 'D39DC0E3'
```
for those having issue with the key just set the key manual when setting up the class in your manifest

```puppet
class { '::rvm': gnupg_key_id => '3804BB82D39DC0E3' }
```

@maestrodev
would you like a pull request?

### LINKED ISSUE #145: Updating rvm signature key
According to [rvm github](https://github.com/rvm/rvm/issues/4520), the releases are now signed with a new key, which prevents new installations using this puppet rules, as the installer can't verify the new signature.

Could you update the key?

Thanks.

### PR DESCRIPTION
**Pull request by saz**
#### Pull Request (PR) description

Fetch gpg signing keys from rvm.io website as not all of them are available via keyservers

#### This Pull Request (PR) fixes the following issues

### CODE REVIEW

File: manifests/system.pp [hunk 1]
```diff
@@ -4,9 +4,8 @@
   $install_from=undef,
   $proxy_url=undef,
   $no_proxy=undef,
-  $key_server=undef,
   $home=$facts['root_home'],
-  $gnupg_key_id=$rvm::params::gnupg_key_id
+  $signing_keys=$rvm::params::signing_keys
```
Comment (bastelfreak) [inline manifests/system.pp:8]: please add a datatype, at least: 
```suggestion
  Hash[String[1], String[1]] $signing_keys=$rvm::params::signing_keys
```

File: manifests/system.pp [hunk 2]
```diff
@@ -76,8 +84,10 @@
   # the fact won't work until rvm is installed before puppet starts
   if $facts['rvm_version'] and !empty($facts['rvm_version']) {
     if ($version != undef) and ($version != present) and ($version != $facts['rvm_version']) {
-      if defined(Class['rvm::gnupg_key']) {
-        Class['rvm::gnupg_key'] -> Exec['system-rvm-get']
+      if $signing_keys {
```
Comment (bastelfreak) [inline manifests/system.pp:87]: if we enforce the hash as datatype, we do not need this line. it is always true

File: manifests/system.pp [hunk 3]
```diff
@@ -76,8 +84,10 @@
...
+        $signing_keys.each |Hash[String[1], String[1]] $key| {
```
Comment (bastelfreak) [inline manifests/system.pp:88]: typo?
```suggestion
        $signing_keys.each |Hash[String[1], String[1] $key| {
```

File: manifests/system.pp (hunk 2 above)
Comment (saz) [reply manifests/system.pp:87]: With the current module version, it's possible to set `gnupg_key_id` to `false` to disable key management. 
I was thinking about to drop this, but there might be valid reasons, to manage keys in a different way.

File: manifests/system.pp (hunk 3 above)
Comment (saz) [reply manifests/system.pp:88]: Looks good to me `Hash[ String[1], String[1] ]`

Comment (bastelfreak) [reply manifests/system.pp:88]: ahh that's the closing bracket from the Hash type

File: manifests/system.pp (hunk 2 above)
Comment (bastelfreak) [reply manifests/system.pp:87]: In that case people could set `$singing_keys` to `{}`

File: manifests/init.pp [hunk 4]
```diff
@@ -10,7 +10,7 @@
   $rvm_gems= {},
   $proxy_url=$rvm::params::proxy_url,
   $no_proxy=$rvm::params::no_proxy,
-  $gnupg_key_id=$rvm::params::gnupg_key_id,
+  Array[Hash[String[1], String[1]]] $signing_keys = $rvm::params::signing_keys,
```
Comment (ekohl) [inline manifests/init.pp:13]: I did the same thing here: https://github.com/voxpupuli/puppet-rvm/pull/18, but I can easily rebase. Yay for data types.

**Review by ekohl — APPROVED** [review]
Thanks! I'll just wait for CI to finish.

**Pull request merged**